handler.go

```diff
@@ -111,6 +114,21 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}()
 
+	for _, mw := range h.middlewares {
+		err := mw.Before(r, h)
+		if err != nil {
+			mwe, ok := err.(MiddlewareError)
+			if ok {
+				w.WriteHeader(mwe.StatusCode)
+			} else {
+				w.WriteHeader(http.StatusInternalServerError)
+			}
+
+			h.errorHandler(w, err)
```

Comment:
I believe ErrorHandler is supposed to write the actual error code.

What we should do here is refactor this to to use the internal exit code handler and not write headers twice (like this does)